src/adapter/src/util.rs

```diff
@@ -71,6 +71,11 @@ impl<T: Transmittable> ClientTransmitter<T> {
             see ClientTransmitter::set_allowed"
         );
 
+        crate::coord::timeout::add_idle_in_transaction_session_timeout(
+            &self.internal_cmd_tx,
+            &session,
+        );
```

Comment:
This does not work. The result often contains a future which is then waited on. So we are not necessarily idle here. great.